pkg/sql/create_index.go

```diff
@@ -425,6 +425,10 @@ func (n *createIndexNode) startExec(params runParams) error {
 	}
 
 	if n.n.Interleave != nil {
+		if n.n.PartitionByIndex != nil {
```

Comment:
oh, you also need to do this for `ALTER TABLE ... PARTITION BY ....`

Reply:
That would partition the primary index, not the secondary index, right? It seems to work fine, from some simple testing.